include/seqan3/argument_parser/validators.hpp

```diff
@@ -485,6 +490,113 @@ struct default_validator
     }
 };
 
+/*!\brief A helper struct to chain validators recursively via the pipe operator.
+ *\ingroup argument_parser
+ *\implements seqan3::validator_concept
+ *
+ *\details
+ *
+ * Note that both validators must operate on the same value_type in order to
+ * avoid unexpected behaviour and ensure that the seqan3::argument_parser::add_option
+ * call is well-formed. (add_option(val, ...., validator) requires
+ * that val is of same type as validator::value_type).
+ */
+template <validator_concept validator1_type, validator_concept validator2_type>
+//!\cond
+    requires std::Same<typename validator1_type::value_type, typename validator2_type::value_type>
+//!\endcond
+class validator_chain_adaptor
+{
+public:
+    //!\brief The underlying type in both validators.
+    using value_type = typename validator1_type::value_type;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief The default constructor is explicitly deleted.
+    validator_chain_adaptor() = delete;
+    validator_chain_adaptor(validator_chain_adaptor const & pf) = default;
+    validator_chain_adaptor & operator=(validator_chain_adaptor const & pf) = default;
+    validator_chain_adaptor(validator_chain_adaptor &&) = default;
+    validator_chain_adaptor & operator=(validator_chain_adaptor &&) = default;
+
+    /*!\brief Constructing from two validators.
+     * \param[in] vali1_ Some validator to be chained to vali2_.
+     * \param[in] vali2_ Another validator to be chained to vali1_.
+     */
+    validator_chain_adaptor(validator1_type const vali1_, validator2_type const vali2_) :
```

Comment:
🤔 The const here prevents the usage of the move constructor for rvalues.

Reply:
:thinking: True... does that mean it should not compile with temporaries? Because it does :eyes:

Reply:
Well it does compile with temporaries, but they aren't moved but copied 🤓